controllers/directional_trading/lp_hedging.py

```diff
@@ -0,0 +1,819 @@
+        default=0,
+        json_schema_extra={
+            "prompt": "Enter the base amount to open the pool (e.g. 100): ",
+            "prompt_on_new": True,
+            "is_updatable": True,
+        },
+    )
+
+    quote_amount: float = Field(
+        default=0,
+        json_schema_extra={
+            "prompt": "Enter the quote amount to open the pool (e.g. 100): ",
+            "prompt_on_new": True,
+            "is_updatable": True,
+        },
+    )
+
+    lower_bound: float = Field(
+        default=0,
+        json_schema_extra={
+            "prompt": "Enter the lower bound for the pool (e.g. 0): ",
+            "prompt_on_new": True,
+            "is_updatable": True,
+        },
+    )
+
+    upper_bound: float = Field(
+        default=0,
+        json_schema_extra={
+            "prompt": "Enter the upper bound for the pool (e.g. 0): ",
+            "prompt_on_new": True,
+            "is_updatable": True,
+        },
+    )
+
+    close_short_on_stop: bool = Field(
+        default=False,
+        json_schema_extra={
+            "prompt": "Close short position on stop (e.g. True): ",
+            "prompt_on_new": True,
+            "is_updatable": True,
+        },
+    )
+
+    close_pool_position_on_stop: bool = Field(
+        default=False,
+        json_schema_extra={
+            "prompt": "Close pool position on stop (e.g. True): ",
+            "prompt_on_new": True,
+            "is_updatable": True,
+        },
+    )
+
+    is_pool_amm: bool = Field(
+        default=False,
+        json_schema_extra={
+            "prompt": "Is the pool an AMM (e.g. True): ",
+            "prompt_on_new": True,
+            "is_updatable": True,
+        },
+    )
+
+
+class LpHedgingError(Exception):
+    pass
+
+
+class LpHedgingNotReady(LpHedgingError, ConnectionError):
+    pass
+
+
+class PoolNotExisting(LpHedgingError, ValueError):
+    pass
+
+
+def calculate_apr_apy(
+    initial_value: Decimal,
+    current_value: Decimal,
+    time_period_sec: float,
+) -> tuple[Decimal, Decimal]:
+    """
+    Calculate APR and APY (assuming continuous compounding) based on initial and current values over a time period.
+
+    Args:
+        initial_value (Decimal): The starting principal amount.
+        current_value (Decimal): The ending amount after interest.
+        time_period_sec (float): Elapsed time in seconds.
+
+    Returns:
+        tuple[Decimal, Decimal]: A tuple containing:
+            - APR as a percentage (Decimal)
+            - APY as a percentage (Decimal, continuous compounding)
+    """
+    # Convert seconds in a year to Decimal
+    # Compute simple rate over period
+    rate_over_period = (current_value / initial_value) - Decimal(1)
+
+    # APR: annualize the simple rate
+    apr = (
+        rate_over_period
+        * 365
+        / (Decimal(time_period_sec) / (Decimal(24) * Decimal(3600)))
+    ) * Decimal(100)
+
+    # APY: continuous compounding: exp(apr_fraction) - 1, then percentage
```

Comment:
I recommend using a simple, non-compounded approach for APR since it's difficult to predict the compounding frequency. Also, CLMM LP position APRs tend to be high anyway, so continuous compounding might overinflate expectations.